mlflow/pytorch/_lightning_autolog.py

```diff
@@ -287,6 +290,109 @@ def on_test_end(self, trainer, pl_module):
+        if self.last_monitor_value is None:
+            return True
+
+        if self.mode == "min":
+            return new_monitor_value <= self.last_monitor_value
+
+        if self.mode == "max":
+            return new_monitor_value >= self.last_monitor_value
+
+        assert False, "Illegal __MLflowModelCheckpoint config."
+
+    def on_train_epoch_end(self, trainer: "pl.Trainer", pl_module: "pl.LightningModule") -> None:
+        current_epoch = trainer.current_epoch
+        metric_dict = {k: float(v) for k, v in trainer.callback_metrics.items()}
+
+        should_checkpoint = False
+        if self.every_n_epochs and (current_epoch % self.every_n_epochs == 0):
+            should_checkpoint = True
+        elif (
+                self.train_time_interval_S and
+                time.time() - self.latest_checkpoint_timestamp > self.train_time_interval_S
+        ):
+            should_checkpoint = True
+
+        if not should_checkpoint:
+            return
+
+        if self.save_best_only:
+            if self.monitor not in metric_dict:
+                # "save-best-only" requires comparing the monitor metric value,
+                # but the provided monitor metric is not available,
+                # skip model checkpoint autologging
+                return
+
+            new_monitor_value = metric_dict[self.monitor]
+            if not self._is_new_checkpoint_better(new_monitor_value):
+                # Current checkpoint is worse than last saved checkpoint,
+                # so skip checkpointing.
+                return
+
+            self.last_monitor_value = new_monitor_value
+
+        if self.save_best_only:
+            if self.save_weights_only:
+                checkpoint_model_filename = "latest_checkpoint_model.weights.pth"
+            else:
+                checkpoint_model_filename = "latest_checkpoint_model.pth"
+            checkpoint_metrics_filename = "latest_checkpoint_metrics.json"
+            checkpoint_artifact_dir = ""
+        else:
+            if self.save_weights_only:
+                checkpoint_model_filename = f"checkpoint_model_epoch_{current_epoch}.weights.pth"
+            else:
+                checkpoint_model_filename = f"checkpoint_model_epoch_{current_epoch}.pth"
+            checkpoint_metrics_filename = f"checkpoint_metrics_epoch_{current_epoch}.json"
+            checkpoint_artifact_dir = "checkpoints"
+
+        mlflow.set_tag(
+            _BEST_CHECKPOINT_ARTIFACT_TAG_KEY,
+            os.path.join(checkpoint_artifact_dir, checkpoint_model_filename)
+        )
+
+        mlflow.log_dict(
+            {**metric_dict, "epoch": current_epoch},
+            os.path.join(checkpoint_artifact_dir, checkpoint_metrics_filename)
+        )
+
+        tmp_dir = create_tmp_dir()
+        try:
+            tmp_model_save_path = os.path.join(tmp_dir, checkpoint_model_filename)
+            trainer.save_checkpoint(tmp_model_save_path, weights_only=self.save_weights_only)
```

Comment:
yes, resuming training from a saved checkpoint is a common use case. Also users are not necessarily loading the latest checkpoint, so I thnk we can provide a public API `mlflow.pytorch.load_checkpoint`, which can load pytorch checkpoint, which includes model weights and optimizer states. This can also be used for vanilla pytorch workflow, not necessarily lightning workflows.

Reply:
oh, the new version API becomes:
```
trainer.fit(model, ckpt_path="some/path/to/my_checkpoint.ckpt")
```
So we can't return a trainer directly without `fit` invocation.

https://lightning.ai/docs/pytorch/stable/common/checkpointing_basic.html#resume-training-state

Reply:
Got it. My previous comment was like we can also load the model weights specifically without relying on the `trainer` instance, which means:
```
model = mlflow.pytorch.load_checkpoint(model_class, mlflow_uri)
trainer = trainer(model)
```
For example, if we are sharing this model, and someone using vanilla pytorch wants to finetune it with a custom training loop, then they can load the checkpoints.